decbinhex4/addons/main/src/code/convert.nim

```diff
@@ -2,28 +2,24 @@ import math, strutils
 import std/times #benchmark only
 
 # the asterisk is so it can be called from another file when imported
-proc convert_from_decimal*(input_decimal: string): string =
-  var result_binary: string = "reset binary result"
-  var result_hex: string = "reset hex result"
-  if input_decimal.len < 20:
-    result_binary = strip((input_decimal.parseInt).toBin(63), trailing = false, chars = {'0'})
+proc convert_from_decimal*(input_decimal: string):string =
+  var result_binary:string; var result_hex:string
+  if input_decimal.len < 20: result_binary = strip((input_decimal.parseInt).toBin(63), trailing = false, chars = {'0'})
   else: result_binary = "Output binary too large!"
   result_hex = strip(input_decimal.parseInt.toHex, trailing = false, chars = {'0'})
   return("Binary is: " & result_binary & "\nHexadecimal is: " & result_hex)
```

Comment:
Is there a reason why you use var at the head of the definition in situations where let can be used?

```nim
proc convert_from_decimal*(input_decimal: string):string =
  let result_binary: string =
    if input_decimal.len < 20:
      strip((input_decimal.parseInt).toBin(63), trailing = false, chars = {'0'})
    else:
      "Output binary too large!"
  let result_hex: string = strip(input_decimal.parseInt.toHex, trailing = false, chars = {'0'})
  return("Binary is: " & result_binary & "\nHexadecimal is: " & result_hex)
```

Reply:
You're right some of those weren't needed and I changed them now.

Others are either accessed multiple times and/or directly used as storage, like result_decimal.